src/components/GraphFilter/GraphFilter.tsx

```diff
@@ -55,16 +74,12 @@ export class GraphFilter extends React.Component<GraphFilterProps, GraphFilterSt
       <div>
         <ButtonToolbar>
           <ButtonGroup>
-            <DropdownButton
-              id="namespace-selector"
-              title={this.props.activeNamespace.name}
-              onSelect={this.updateNamespace}
-            >
-              {this.state.availableNamespaces.map(ns => (
-                <MenuItem key={ns.name} active={ns.name === this.props.activeNamespace.name} eventKey={ns.name}>
-                  {ns.name}
-                </MenuItem>
-              ))}
+            <DropdownButton id="namespace-selector" title={this.props.activeNamespace.name} onClick={this.reloadData}>
+              <DropdownList
+                list={namespaceStore.namespaceList}
```

Comment:
Question, should we move this list to a props? `list={this.props.namespaces}`? That would make it stateless (desired), right?

Reply:
Ideally yes. This direct “sideway” injection (as oppose to “top down” via props) is a temp solution due to difficulty I had with typescript as mentioned in the PR foot note.